### PR TITLE
[MRG] Refactor utility functions useful in testing

### PR DESCRIPTION
**Pull request by kchawla-pi**
- Functions moved from /tests/test_first_level_model.py ino /nistats/utils.py:
   - write_fake_fmri_data()
   - generate_fake_fmri_data()
 - Functions moved from /tests/test_utils.py into /nistats/utils.py (some have been renamed):
   - write_fake_bold_img()  -->  _write_fake_bold_img()  
   - basic_paradigm()  -->  _basic_paradigm()
   - basic_confounds()  -->  _basic_confounds()
   - create_fake_bids_dataset()
 - Duplicate function removed from /tests/test_second_level_model.py:
   - write_fake_fmri_data()

### CODE REVIEW

**Comment by bthirion** [comment]
I'm not sure I understand your goal. Since these functions are used only for testing, why not simply keep them in the test directory ?

**Comment by kchawla-pi** [comment]
Nilearn has it organized thusly: `nilearn/_utils/testing.py` . I can do the same for nistats, and eventually (in a separate PR) move the `utils.py` file into `\_utils` as well. We have to start reorganizaing nistats to be Nilearn compatible anyway.

**Comment by bthirion** [comment]
Indeed. Please refactor accordingly.

**Comment by kchawla-pi** [comment]
Okay, once this is merged I will open a new PR to move the existing `nistats/utils.py` into `nistats/_utils/utils.py`.

File: nistats/_utils/testing.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    data = np.random.rand(length, 6)
+    confounds = pd.DataFrame(data, columns=columns)
+    return confounds
+
+
+def create_fake_bids_dataset(base_dir='', n_sub=10, n_ses=2,
+                             tasks=['localizer', 'main'],
+                             n_runs=[1, 3], with_derivatives=True,
+                             with_confounds=True, no_session=False):
+    """Returns a fake bids dataset directory with dummy files
```
Comment (bthirion) [inline nistats/_utils/testing.py]: prepend an underscore to the function name or write a better docstring ;-)

File: nistats/tests/test_first_level_model.py [hunk 2]
```diff
@@ -32,45 +32,18 @@
                                        run_glm,
                                        )
 
-
-from nistats.tests.test_utils import create_fake_bids_dataset
-from nistats.utils import get_bids_files
-
+from nistats.utils import (get_bids_files,
+                           )
```
Comment (bthirion) [inline nistats/tests/test_first_level_model.py]: Maybe I'm wrong but I really prefer the more natural
```
from nistats.utils import get_bids file
```

Comment (kchawla-pi) [reply nistats/tests/test_first_level_model.py]: That is an oversight, the parentheses are not desired for a single entry.

File: nistats/_utils/testing.py (hunk 1 above)
Comment (kchawla-pi) [reply nistats/_utils/testing.py]: :-D
Understood

Comment (kchawla-pi) [reply nistats/_utils/testing.py]: Should I do both prepend with underscore and improve the docstring?

Comment (bthirion) [reply nistats/_utils/testing.py]: I wouldn't say 'you should' but that's even better if you do both.

**Pull request merged**